OrderCloud.Integrations.Payment.Stripe/Mappers/StripePaymentIntentMapper.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using OrderCloud.Catalyst;
+using Stripe;
+
+namespace OrderCloud.Integrations.Payment.Stripe.Mappers
+{
+    /// <summary>
+    /// https://stripe.com/docs/api/payment_intents
+    /// </summary>
+    public class StripePaymentIntentMapper
+    {
+        public PaymentIntentCreateOptions MapPaymentIntentCreateAndConfirmOptions(AuthorizeCCTransaction transaction) =>
+            new PaymentIntentCreateOptions()
+            {
+                Amount = Convert.ToInt64(transaction.Amount),
+                Confirm = true, // Creates and Confirms PaymentIntent, otherwise Confirm PaymentIntent would be a separate call
+                CaptureMethod = "manual", // Required value for separate auth and capture
+                Currency = transaction.Currency,
+                Customer = transaction.ProcessorCustomerID,
+                PaymentMethod = transaction.TransactionID, // Represents PaymentMethodID
```

Comment:
We talked about `PaymentMethod = transaction.TransactionID` being different than the other providers. I just wanted to make sure there was a way to support both 1 time cards and saved cards in this integration. The `AuthorizeCCTransaction` model has .cardDetails.Token and .cardDetails.SavedCardID. Not sure what step in Stripe needs that info though.

Reply:
The flow for using a saved card would be for the user's middleware or client to list PaymentMethods for a customer where type is "card", and then use the ID of the selected PaymentMethod. Here is a unit test that succeeds demonstrating the flow:
```[Test]
        public async Task CreateCustomerAndCard_UseSavedCard()
        {
            var paymentSysCustomer = new PaymentSystemCustomer()
            {
                FirstName = "Test",
                LastName = "User",
                Email = "testuser@sitecore.com",
                CustomerAlreadyExists = false
            };

            var pciSafeCardDetails = new PCISafeCardDetails()
            {
                Token = "tok_mastercard" // this is specific for Stripe testing, otherwise it would be a card object
            };

            // Create Card AND Customer
            var createdCard = await _service.CreateSavedCardAsync(paymentSysCustomer, pciSafeCardDetails);
            Assert.IsNotNull(createdCard.Card.Token);

            // List Customer Payment Methods
            var paymentMethods = await _client.ListPaymentMethodsAsync(new PaymentMethodListOptions()
            {
                Customer = createdCard.CustomerID,
                Type = "card"
            });

            Assert.AreEqual(createdCard.Card.Token, paymentMethods.Data.FirstOrDefault().Card.Fingerprint);

            var transaction = new AuthorizeCCTransaction()
            {
                Amount = 500,
                Currency = "USD",
                ProcessorCustomerID = createdCard.CustomerID,
                TransactionID = paymentMethods.Data.FirstOrDefault().Id
            };

            var paymentIntent = await _service.AuthorizeOnlyAsync(transaction, _config);
            Assert.AreEqual("requires_capture", paymentIntent.Message);

            var captureTransaction = new FollowUpCCTransaction()
            {
                Amount = 500,
                TransactionID = paymentIntent.TransactionID
            };
            var capturePaymentIntent = await _service.CapturePriorAuthorizationAsync(captureTransaction, _config);
            Assert.AreEqual("succeeded", capturePaymentIntent.Message);
        }```
https://stripe.com/docs/api/payment_methods/create#create_payment_method-card

Reply:
The update to use PaymentMethods looks great! Could we change this line to something like
`PaymentMethod = transaction?.CardDetails?.SavedCardID ?? transaction.TransactionID,`?
Just so it uses that SavedCardID if its set.